src/Rules/IssetCheck.php

```diff
@@ -143,6 +145,23 @@ public function check(Expr $expr, Scope $scope, string $operatorDescription, str
 			}
 
 			if ($propertyReflection->hasNativeType() && !$propertyReflection->isVirtual()->yes()) {
+				if (
+					$expr instanceof Node\Expr\PropertyFetch
+					&& $expr->name instanceof Node\Identifier
+					&& $expr->var instanceof Expr\Variable
+					&& $expr->var->name === 'this'
+					&& !TypeCombinator::containsNull($propertyReflection->getNativeType())
```

Comment:
1) Doing `Type::isNull()` would make more sense. This example doesn't cover `mixed` (which can include null).
2) But I think we should pay more attention to `$typeMessageCallback` passed to IssetCheck. For both the type check and the error message. Something a little bit different happens in EmptyRule which should be covered by a set of tests. (Because `empty($x)` is `!isset($x) || $x == false`).